### PR TITLE
Refactored NoCachingTransactionSynchronizationImpl to avoid timestamp…

### PR DESCRIPTION
**Pull request by dreab8**
… creation, Renamed CacheTransactionSynchronization#getCurrentTransactionStartTimestamp method to getCachingTimestamp and removed SharedSessionContractImplementor#getTransactionStartTimestamp method

https://hibernate.atlassian.net/browse/HHH-15502

### CODE REVIEW

**Comment by Sanne** [comment]
Thanks @dreab8 , I think that's a good approach.
I'll comment inline on a couple of clarifications, but I think we could then merge it.

File: hibernate-core/src/main/java/org/hibernate/cache/spi/CacheTransactionSynchronization.java [hunk 1]
```diff
@@ -48,8 +48,10 @@ public interface CacheTransactionSynchronization {
 	 *
 	 * @implSpec This "timestamp" need not be related to timestamp in the Java
 	 * Date/millisecond sense.  It just needs to be an incrementing value.
+	 *
+	 * An UnsupportedOperationException is thrown for No Caching
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/cache/spi/CacheTransactionSynchronization.java:52]: I'm not sure if it's clear what "No Caching" means. Could you clarify that?

File: hibernate-core/src/main/java/org/hibernate/cache/internal/NoCachingTransactionSynchronizationImpl.java [hunk 2]
```diff
@@ -6,14 +6,35 @@
...
+public class NoCachingTransactionSynchronizationImpl implements CacheTransactionSynchronization {
+	public static final NoCachingTransactionSynchronizationImpl INSTANCE = new NoCachingTransactionSynchronizationImpl();
+
+	private NoCachingTransactionSynchronizationImpl() {
+
+	}
+
+	@Override
+	public long getCachingTimestamp() {
+		throw new UnsupportedOperationException("No Caching does not support Timestamp generation");
```
Comment (Sanne) [inline hibernate-core/src/main/java/org/hibernate/cache/internal/NoCachingTransactionSynchronizationImpl.java:23]: I'm not sure if it's clear what "No Caching" means. Could you clarify that?

**Review by Sanne — CHANGES_REQUESTED** [review]

**Comment by Sanne** [comment]
> Thanks @dreab8 , I think that's a good approach. I'll comment inline on a couple of clarifications, but I think we could then merge it.

we'll also need a JIRA, of course.

**Comment by Sanne** [comment]
@dreab8 could you also add this change please:
 - https://github.com/hibernate/hibernate-orm/commit/ab58157cae10da81c421ffe453983a9179d6a50c

File: hibernate-core/src/main/java/org/hibernate/cache/internal/NoCachingTransactionSynchronizationImpl.java (hunk 2 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/cache/internal/NoCachingTransactionSynchronizationImpl.java]: changed, what do you think? any better?

File: hibernate-core/src/main/java/org/hibernate/cache/spi/CacheTransactionSynchronization.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/cache/spi/CacheTransactionSynchronization.java]: changed, what do you think?

**Pull request closed without merging**